grammar.js

```diff
@@ -317,25 +317,38 @@ grammar({
     // Statements
 
     _statement: $ => choice(
+      $.compound_statement,
```

Comment:
What is the reason for the renaming? Aren't `begin ... end` block technically expressions? Same with `let` and `if` (and probably more), these are also expressions, right?

Reply:
Yes, they are. Julia is an expression based language and all forms are expressions that return some value.

`_statement` does conflate a few of things:
1. Forms that hold blocks, like `quote … end`
2. Forms that "never return" a value like `return` and `break`
3. Forms that can only be at the top-level, like `import`

I would like to have a separate name for the first one, because we have `quote_statement` for `quote … end` and `quote_expression` for `:(…)`. Similarly, we'd have `compound_statement` and `parenthesized_expression`, which aren't distinguished in the IR (they both form block Exprs). I think this is more consistent.

I have no opinion on the second category.

I don't think the third one should have any special treatment, because you can write those expressions inside quotes (which are not necessarily top-level).

`_statement` is a "subtype" of `_expression` in the grammar, so this more of a naming problem than a parsing problem.